### PR TITLE
190 feature/delete posts by custom taxonomy

### PR DESCRIPTION
**Pull request by rajanit2000**

### CODE REVIEW

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace BulkWP\BulkDelete\Core\Posts\Metabox;
+
+use BulkWP\BulkDelete\Core\Posts\PostsMetabox;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Delete Posts by Status Metabox.
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: Comment is wrong. Looks like a copy/paste error.

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		}
+
+		if ( count( $terms_array ) > 0 ) {
+?>
+        <!-- Custom tax Start-->
+        <h4><?php _e( 'Select the post type from which you want to delete posts by custom taxonomy', 'bulk-delete' ); ?></h4>
+
+        <fieldset class="options">
+            <table class="optiontable">
+				<?php bd_render_post_type_dropdown( 'tax' ); ?>
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: Use the `render_post_type_dropdown` function from `PostsMetabox` class

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+<?php
+		}
+	}
+
+	protected function convert_user_input_to_options( $request, $options ) {
+		$options                       = array();
+		$options['post_type']          = bd_array_get( $request, 'smbd_tax_post_type', 'post' );
+		$options['selected_taxs']      = bd_array_get( $request, 'smbd_taxs' );
+		$options['selected_tax_terms'] = bd_array_get( $request, 'smbd_tax_terms' );
+		$options['restrict']           = bd_array_get_bool( $request, 'smbd_taxs_restrict', false );
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: Apart from `post_type`, `selected_taxs` and `selected_tax_terms`, you can remove the others.

They are handled in the `process_common_filters` method.

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		$options['force_delete']       = bd_array_get_bool( $request, 'smbd_taxs_force_delete', false );
+
+		$options['date_op']            = bd_array_get( $request, 'smbd_taxs_op' );
+		$options['days']               = absint( bd_array_get( $request, 'smbd_taxs_days' ) );
+
+		return $options;
+	}
+
+	public function delete( $delete_options ) {
+		if ( bd_array_get( $_POST, 'smbd_taxs_cron', 'false' ) == 'true' ) {
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: You don't have to handle cron here. It is already handled. Check other metaboxes and also the `process_cron_filters` method.

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+				wp_schedule_event( $time, $freq, Bulk_Delete::CRON_HOOK_TAXONOMY, array( $delete_options ) );
+			}
+			$msg = __( 'Posts from the selected custom taxonomies are scheduled for deletion.', 'bulk-delete' ) . ' ' .
+				sprintf( __( 'See the full list of <a href = "%s">scheduled tasks</a>' , 'bulk-delete' ), get_bloginfo( 'wpurl' ) . '/wp-admin/admin.php?page=' . Bulk_Delete::CRON_PAGE_SLUG );
+		} else {
+			return $deleted_count = $this->delete_posts_by_taxonomy( $delete_options );
+		}
+	}
+
+	public static function delete_posts_by_taxonomy( $delete_options ) {
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: This should not be either public or static.

Also, we can replace this method with just the build_query method.

Look into either DeletePostsByCategoryMetabox or DeletePostsByTagMetabox to see how to do it. This is something that I recently added.

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php (hunk 3 above)
Comment (rajanit2000) [reply include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: @sudar this code have some cnflicts, I could not merge it using my git GUI enveronent. I hope it ll be fixed after merging

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+use BulkWP\BulkDelete\Core\Posts\PostsMetabox;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Delete Posts by Cudtom Taxonomy Metabox.
+ *
+ * @since 6.0.0
+ */
+class DeletePostsByCustomTaxonomMetabox extends PostsMetabox {
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: There is a typo in the class name and file name.

File: include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php [hunk 7]
```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace BulkWP\BulkDelete\Core\Posts\Metabox;
+
+use BulkWP\BulkDelete\Core\Posts\PostsMetabox;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Delete Posts by Cudtom Taxonomy Metabox.
```
Comment (sudar) [inline include/Core/Posts/Metabox/DeletePostsByCustomTaxonomMetabox.php]: There is a typo in the comment.

**Pull request merged**